api/router/exam.py

```diff
@@ -19,14 +21,39 @@
 """,
 )
 def get_question(
-    question_number: int,
-    assessment: Assessment = Depends(get_assessment),
+        question_number: int,
+        assessment: Assessment = Depends(get_assessment),
 ):
     if question_number not in assessment.questions:
         raise HTTPException(status_code=HTTP_404_NOT_FOUND, detail="Question not found")
     return assessment.questions[question_number]
 
 
+# ------------------------------------------ Answers
+
+
+@exam_router.get(
+    "/questions/{question_number}/answer",
+    tags=["answer"],
+    response_model=list[AnswerRead],
+    summary="Retrieve answer for question by user",
+    description="""
+Retrieve the given latest answer by a user including part, section, task.
+""",
+)
+def get_answer(
+        question_number: int,
+        session: Session = Depends(get_session),
+
```

Comment:
We want at least 

```suggestion
assessment_id: str = Depends(get_assessment_id)
```

and then filter answers by that (`exam_id=assessment_id`). Otherwise we'd be querying this student's answer to ALL their assessments.

Reply:
Noted and added